libs/pricing_model/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+/// Maintains latest [`Segment`] for each of the branches, faciliates the pricing model
+/// calculations.
```

Comment:
The `segments` vector maintains *all* segments, not just the latest of each branch. This comment refers just to the 'branches` field, not the whole struct.

Reply:
Enhanced this in [32322b9](https://github.com/neondatabase/neon/pull/2714/commits/32322b9086a8708b2c6b39393fa575a9c0505aa5)